new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsShutdown.java

```diff
@@ -0,0 +1,290 @@
+                .replicas(replicaCount)
+                .serverStartState("RUNNING")
+                .serverPod(new ServerPod().shutdown(new Shutdown().timeoutSeconds(80L)))
+                .serverPod(new ServerPod().shutdown(new Shutdown().ignoreSessions(true))))
+            .configuration(new Configuration()
+                .model(new Model()
+                    .domainType(WLS_DOMAIN_TYPE)
+                    .runtimeEncryptionSecret(encryptionSecretName)))
+            .addManagedServersItem(new ManagedServer()
+                .serverName(managedServer1Name)
+                .serverPod(new ServerPod().shutdown(new Shutdown().timeoutSeconds(100L)))));
+
+    // create model in image domain
+    logger.info("Creating model in image domain {0} in namespace {1} using docker image {2}",
+        domainUid, domainNamespace, miiImage);
+    createDomainAndVerify(domain, domainNamespace);
+
+    // check that admin server pod exists in the domain namespace
+    logger.info("Checking that admin server pod {0} exists in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodExists(adminServerPodName, domainUid, domainNamespace);
+
+    // check that admin server pod is ready
+    logger.info("Checking that admin server pod {0} is ready in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+
+    // check that admin service exists in the domain namespace
+    logger.info("Checking that admin service {0} exists in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
+
+    // check for managed server pods existence in the domain namespace
+    for (int i = 1; i <= replicaCount; i++) {
+      String managedServerPodName = managedServerPrefix + i;
+
+      // check that the managed server pod exists in the domain namespace
+      logger.info("Checking that managed server pod {0} exists in namespace {1}",
+          managedServerPodName, domainNamespace);
+      checkPodExists(managedServerPodName, domainUid, domainNamespace);
+
+      // check that the managed server pod is ready
+      logger.info("Checking that managed server pod {0} is ready in namespace {1}",
+          managedServerPodName, domainNamespace);
+      checkPodReady(managedServerPodName, domainUid, domainNamespace);
+
+      // check that the managed server service exists in the domain namespace
+      logger.info("Checking that managed server service {0} exists in namespace {1}",
+          managedServerPodName, domainNamespace);
+      checkServiceExists(managedServerPodName, domainNamespace);
+    }
+  }
+
+  /**
+   * Verify the server pod Shutdown properties.
+   */
+  private static boolean verifyServerShutdownProp(
+      String podName,
+      String domainNS,
+      String... props) {
+
+    StringBuffer cmd = new StringBuffer("kubectl get pod ");
+    cmd.append(podName);
```

Comment:
should we change it to k8s API?

Reply:
Do you mean using something like
Domain domain1 = getDomainCustomResource(domainUid, domainNamespace) then from there to domain1.get(xx).get(xx)...
If yes, it would be the same as it was originally defined. I might misunderstand you though.

Reply:
no I meant to use this:
V1Pod managedServerPod = Kubernetes.getPod(domainNS, null, managedServerPodName);

    //check that managed server pod is up and all applicable variable values are initialized.
    assertNotNull(managedServerPod,"The managed server pod does not exist in namespace " + domainNS);
    V1ObjectMeta managedServerMetadata = managedServerPod.getMetadata();
    String serverName = managedServerMetadata.getLabels().get("servername");

you can retrieve all data from there

Reply:
Sure. Let me change it

Reply:
Done. Please review. Thanks.